docs/vec_of_option.md

```diff
@@ -0,0 +1,212 @@
+{
+    // This implementation differs from pandas as that boundary None's are not removed.
+    // This prevents a lot of errors due to expressions leading to different lengths.
+    if !chunked_arr.has_nulls() || chunked_arr.null_count() == chunked_arr.len() {
+        return chunked_arr.clone();
+    }
+
+    // We first find the first and last so that we can set the null buffer.
+    let first = chunked_arr.first_non_null().unwrap();
+    let last = chunked_arr.last_non_null().unwrap() + 1;
+
+    // Fill out with `first` nulls.
+    let mut out = Vec::with_capacity(chunked_arr.len());
+    let mut iter = chunked_arr.iter().skip(first);
+    for _ in 0..first {
+        out.push(Zero::zero());
+    }
+
+    // The next element of `iter` is definitely `Some(Some(v))`, because we skipped the first
+    // elements `first` and if all values were missing we'd have done an early return.
+    let mut low = iter.next().unwrap().unwrap();
+    out.push(low);
+    while let Some(next) = iter.next() {
+        if let Some(v) = next {
+            out.push(v);
+            low = v;
+        } else {
+            let mut steps = 1 as IdxSize;
+            for next in iter.by_ref() {
+                steps += 1;
+                if let Some(high) = next {
+                    let steps_n: T::Native = NumCast::from(steps).unwrap();
+                    interpolation_branch(low, high, steps, steps_n, &mut out);
+                    out.push(high);
+                    low = high;
+                    break;
+                }
+            }
+        }
+    }
+    if first != 0 || last != chunked_arr.len() {
+        let mut validity = MutableBitmap::with_capacity(chunked_arr.len());
+        validity.extend_constant(chunked_arr.len(), true);
+
+        for i in 0..first {
+            validity.set(i, false);
+        }
+
+        for i in last..chunked_arr.len() {
+            validity.set(i, false);
+            out.push(Zero::zero())
+        }
+
+        let array = PrimitiveArray::new(
+            T::get_dtype().to_arrow(CompatLevel::newest()),
+            out.into(),
+            Some(validity.into()),
+        );
+        ChunkedArray::with_chunk(chunked_arr.name(), array)
+    } else {
+        ChunkedArray::from_vec(chunked_arr.name(), out)
+    }
+}
+```
+
+That's a lot to digest at once, so let's take small steps and focus on the core logic.
+At the start, we store the indexes of the first and last null values:
+
+```rust
+let first = chunked_arr.first_non_null().unwrap();
+let last = chunked_arr.last_non_null().unwrap() + 1;
+```
+
+Then, we push actual zeroes to the output `Vec` - as many as there are nulls before the first valid value in the input:
+
+```rust
+let mut out = Vec::with_capacity(chunked_arr.len());
+for _ in 0..first {
+    out.push(Zero::zero());
+}
+```
+
+We skip the first `first` elements and start interpolating, but the droids we're looking for are not here, so let's skip that part:
```

Comment:
> the droids we're looking for are not here

what does this mean?

Reply:
ah looks like it's a Star Wars phrase 😄 i guess most people will know it, so ok to keep in